protocol-tokio/src/protocol/accepting.rs

```diff
@@ -9,39 +9,48 @@ use tokio_io::{AsyncRead, AsyncWrite};
 use cbor_event::de::Deserializer;
 use std::{self, io::Cursor, vec};
 
-use super::{nt, Connection, Handshake, Message, NodeId};
+use super::{nt, raw_msg_to_nt, Connection, Handshake, Message, NodeId};
 
-enum AcceptingState<T> {
+enum AcceptingState<T, Header, BlockId, Block, TransactionId> {
```

Comment:
OK... I'm afraid I'll have to do something like this in network-grpc, but I'm trying to resolve it in terms of chain-core property traits.